tests/pytest_tests/unit_tests/test_public_api.py

```diff
@@ -7,7 +7,8 @@
 from wandb.sdk.internal.thread_local_settings import _thread_local_api_settings
 
 
-def test_api_auto_login_no_tty():
+def test_api_auto_login_no_tty(monkeypatch):
+    monkeypatch.setattr("sys.stdin", None)
```

Comment:
would you mind using a contextmanager  and `unittest.mock` instead?

Reply:
done!